Mathlib/Analysis/Analytic/Order.lean

```diff
@@ -90,6 +90,31 @@ lemma apply_eq_zero_of_order_toNat_ne_zero (hf : AnalyticAt 𝕜 f z₀) :
   simp [hf.order_eq_zero_iff]
   tauto
 
+/-- Characterization of which natural numbers are `≤ hf.order`. Useful for avoiding case splits,
+since it applies whether or not the order is `∞`. -/
+lemma natCast_le_order_iff (hf : AnalyticAt 𝕜 f z₀) {n : ℕ} :
+    n ≤ hf.order ↔ ∃ g, AnalyticAt 𝕜 g z₀ ∧ ∀ᶠ z in 𝓝 z₀, f z = (z - z₀) ^ n • g z := by
+  unfold order
+  split_ifs with h
+  · simpa using ⟨0, analyticAt_const .., by simpa⟩
+  · let m := (hf.exists_eventuallyEq_pow_smul_nonzero_iff.mpr h).choose
+    obtain ⟨g, hg, hg_ne, hm⟩ : ∃ g, AnalyticAt 𝕜 g z₀ ∧ g z₀ ≠ 0 ∧
+        ∀ᶠ z in 𝓝 z₀, f z = (z - z₀) ^ m • g z :=
+      (hf.exists_eventuallyEq_pow_smul_nonzero_iff.mpr h).choose_spec
```

Comment:
```suggestion
    obtain ⟨g, hg, hg_ne, hm⟩ := (hf.exists_eventuallyEq_pow_smul_nonzero_iff.mpr h).choose_spec
```

Reply:
Trivial change that saves us two lines. Feel free to ignore.

Reply:
Sure. I generally include the type explicitly when using `have` or `obtain` with a proof term, even when Lean can infer it, because it makes things easier for human readers. However, in this case the types of `hg` etc are pretty clear already from the name of the lemma `exists_eventuallyEq_pow_smul_nonzero_iff` that's being called, so the type annotation is indeed redundant.